codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/Paginators.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.go.codegen.integration;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.go.codegen.SymbolUtils;
+import software.amazon.smithy.go.codegen.TriConsumer;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.knowledge.PaginatedIndex;
+import software.amazon.smithy.model.knowledge.PaginationInfo;
+import software.amazon.smithy.model.knowledge.TopDownIndex;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.traits.DocumentationTrait;
+
+/**
+ * Implements support for PaginatedTrait.
+ */
+public class Paginators implements GoIntegration {
+    @Override
+    public void writeAdditionalFiles(
+            GoSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            TriConsumer<String, String, Consumer<GoWriter>> writerFactory
+    ) {
+        ServiceShape serviceShape = settings.getService(model);
+
+        PaginatedIndex paginatedIndex = PaginatedIndex.of(model);
+
+        TopDownIndex topDownIndex = TopDownIndex.of(model);
+
+        List<PaginationInfo> paginationInfos = topDownIndex.getContainedOperations(serviceShape).stream()
+                .map(operationShape -> paginatedIndex.getPaginationInfo(serviceShape, operationShape))
+                .filter(Optional::isPresent)
+                .map(paginationInfo -> paginationInfo.get())
+                .collect(Collectors.toList());
+
+        if (paginationInfos.size() == 0) {
+            return;
+        }
+
+        writerFactory.accept("paginators.go", settings.getModuleName(), writer -> {
+            paginationInfos.forEach((paginationInfo) -> {
+                generateOperationPaginator(model, symbolProvider, writer, paginationInfo);
+            });
+        });
```

Comment:
Thoughts on using paginators file vs operation file? Since they are per operation.  Using the operation shape is an easy way to get a writer for shape.